massa-final-state/src/final_state.rs

```diff
@@ -86,9 +86,9 @@ impl FinalState {
     pub fn reset(&mut self) {
```

Comment:
Tradeoff here see the followup in  #3728

Reply:
Why not just recreate the final_state object everytime we attempt a bootstrap ?

Reply:
IIRC I did not find a way with minimal modifications mainly because the final_state is behind a & mut.

Reply:
After analyzing the code. The final state is created before the bootstrap and sent as a `RwLock` to the bootstrap but I don't think it's necessary as it's not used anywhere before getting filled by the bootstrap. So we could create the first one before the bootstrap or inside, and return the filled one to be used in the rest of the code.

As the code that launch the bootstrap is under a `tokio::select` I think it should be better to continue on that PR after bootstrap de-async